src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -66,40 +64,8 @@ public async Task Invoke(HttpContext context)
             // Note that the resource will be null if there is no matched endpoint
             var authorizeResult = await policyEvaluator.AuthorizeAsync(policy, authenticateResult, context, resource: endpoint);
 
-            if (authorizeResult.Challenged)
-            {
-                if (policy.AuthenticationSchemes.Count > 0)
-                {
-                    foreach (var scheme in policy.AuthenticationSchemes)
-                    {
-                        await context.ChallengeAsync(scheme);
-                    }
-                }
-                else
-                {
-                    await context.ChallengeAsync();
-                }
-
-                return;
-            }
-            else if (authorizeResult.Forbidden)
-            {
-                if (policy.AuthenticationSchemes.Count > 0)
-                {
-                    foreach (var scheme in policy.AuthenticationSchemes)
-                    {
-                        await context.ForbidAsync(scheme);
-                    }
-                }
-                else
-                {
-                    await context.ForbidAsync();
-                }
-
-                return;
-            }
-
-            await _next(context);
+            var authorizationMiddlewareResultHandler = context.RequestServices.GetRequiredService<IAuthorizationMiddlewareResultHandler>();
+            await authorizationMiddlewareResultHandler.HandleAsync(_next, context, policy, authorizeResult);
```

Comment:
Oh and another question.
Aspnetcore is not using `SynchronizationContext`, right? So is there a reason why `.ConfigureAwait(false)` is not being used everywhere?

Reply:
This post probably explains it better than I can, https://blog.stephencleary.com/2017/03/aspnetcore-synchronization-context.html but basically yup we don't use ConfigureAwait

Reply:
I see... So just to ensure I got it right:
Currently there's actually no difference whatsoever in ASP .NET Core whether I use ConfigureAwait or not. Is that right?

Reply:
Correct

Reply:
ok thanks :)